content/200-concepts/100-components/03-prisma-migrate/050-prisma-migrate-flows.mdx

```diff
@@ -0,0 +1,234 @@
+
+## Collaborating with a team
+
+To bring in changes to the database schema and migrations from collaborators, pull the changed Prisma Schema and the new migration files. Run `prisma migrate dev` to apply the changes. Running this command:
+
+1. Checks if there are any migrations that have not yet been applied and applies those migrations
+1. Checks for changes in the Prisma schema - if there are changes, Prisma creates and applies new migrations
+
+## Handling migration history conflicts
+
+A migration history conflict occurs when there are discrepancies between the **migrations folder in the file system** and the **`_prisma_migrations` table in the database**. 
+
+### Development environments
+
+In a development environment, history conflicts can occur in the following scenarios:
+
+* A migration that has already been applied is later modified - this check requires a shadow database and is therefore only available in a development environment
+* A migration that has already been applied is missing from the file system
+* A migration failed
+
+If Prisma Migrate detects a migration history conflict when you run `prisma migrate dev` (Development command), the CLI will ask to reset the database. Resetting the database:
+
+1. Drops the database
+2. Recreates the database
+3. Applies all migrations
+
+### Production environments
+
+In a production environment, history conflicts can occur in the following scenarios:
+
+* A migration that has already been applied is missing from the file system
+* A migration failed
+
+If Prisma Migrate detects a migration history conflict when you run `prisma migrate deploy` (Production command)
+
+* Add the missing migration file - for example, from source control
+* Address the cause of the failed migration - each migration in the `_prisma_migrations` table has a `logs` column that stores the error. For example:
+
+  ```
+  Database error: Error accessing result set, column not found: name
+   0: migration_core::api::ApplyMigrations
+             at migration-engine\core\src\api.rs:102
+  ```
+
+## Handling database schema drift
+
+Database schema drift occurs when your database schema is out of sync with your migration history - the database schema has 'drifted away' from the source of truth. Schema drift can occur if:
+
+* The database schema was changed _without_ using migrations - for example, by using [`prisma db push`](../../reference/api-reference/command-reference#db-push) or manually changing the database schema.
+
+> **Note**: Schema drift can only be detected in a development environment as this requires the shadow database.
+
+The following scenario assume that you made a manual change in production and want to propagate that change to your migration history and other databases.
+
+### Development environments
+
+1. Make an equivalent change in the Prisma schema - for example, adding a field named `bio` and an index:
+
+  ```prisma
+  model User {
+      bio   String?
+
+      @@index([bio])
+  }
```

Comment:
Thats technically not a valid prisma model since it has no id and would be invalid for the formatter.